src/components/query-panel/QueryPanelCtrl.ts

```diff
@@ -19,20 +19,28 @@ export class QueryPanelCtrl implements ng.IComponentController {
 
   executeAction(): void {
     this.executionActionHelper().then((response: any) => {
-      this.statusCode = 200;
+      if (isNotNullOrUndefined(response.status)) {
+        this.statusCode = response.status;
+      } else {
+        this.statusCode = 200;
+      }
       if (isNotNullOrUndefined(response) && isNotNullOrUndefined(response.data)) {
         if (isObject(response.data)) {
           this.editor.setValue(JSON.stringify(response.data, null, "\t"));
         } else {
-          this.editor.setValue("Request executed successfully. No response returned.");
+          this.editor.setValue(response.statusText);
         }
       }
     }).catch((error: any) => {
-      this.statusCode = 401;
+      if (isNotNullOrUndefined(error.status)) {
+        this.statusCode = error.status;
+      } else {
+        this.statusCode = 401;
```

Comment:
Why would this be 401? The problem is not related to authorization. See https://en.wikipedia.org/wiki/List_of_HTTP_status_codes